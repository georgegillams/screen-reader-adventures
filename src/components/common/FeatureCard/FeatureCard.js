import React from 'react';
import PropTypes from 'prop-types';
import { FeatureCard as GGFeatureCard, FEATURE_CARD_LAYOUTS } from 'gg-components/FeatureCard';
import nextifyHref from 'client-utils/nextifyHref';
import Link from 'next/link';

const FeatureCard = props => {
  const { href, hrefExternal, onClick, ...rest } = props;

  const renderNormalCard = !href || hrefExternal;

  const destination = nextifyHref(href);

  if (renderNormalCard) {
    return <GGFeatureCard href={href} hrefExternal={hrefExternal} onClick={onClick} {...rest} />;
  }

  return (
    <Link passHref href={destination.url} as={destination.as} {...destination.options}>
      <GGFeatureCard {...rest} />
    </Link>
  );
};

FeatureCard.propTypes = {
  href: PropTypes.string.isRequired,
  hrefExternal: PropTypes.bool,
  scroll: PropTypes.bool,
  onClick: PropTypes.func,
};

FeatureCard.defaultProps = {
  hrefExternal: false,
  scroll: true,
  onClick: null,
};

export default FeatureCard;
export { FEATURE_CARD_LAYOUTS };
