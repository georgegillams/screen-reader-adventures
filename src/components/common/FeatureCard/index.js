export { default, FEATURE_CARD_LAYOUTS } from './FeatureCard';
