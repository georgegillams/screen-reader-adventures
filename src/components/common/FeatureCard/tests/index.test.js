import React from 'react';

import { render } from '@testing-library/react';

import FeatureCard from '..';

describe('<NavigationItem />', () => {
  it('Should render correctly', () => {
    const { container } = render(<FeatureCard href="/test" />);

    expect(container).toMatchSnapshot();
  });
  it('Should render with nextified href', () => {
    const { container } = render(<FeatureCard href="/blog/test" />);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with custom className', () => {
    const { container } = render(<FeatureCard href="/test" className={'test-custom-classname'} />);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly hrefExternal', () => {
    const { container } = render(<FeatureCard href="https://duckduckgo.com/" />);

    expect(container).toMatchSnapshot();
  });
});
