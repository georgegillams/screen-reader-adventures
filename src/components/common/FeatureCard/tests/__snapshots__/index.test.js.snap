// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> Should render correctly 1`] = `
<div>
  <a
    class="card feature-card card--atomic card--not-disabled"
    href="/test"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      >
        <div
          aria-hidden="true"
          class="feature-card__outer-container feature-card__outer-container--auto"
        >
          <div
            class="feature-card__date"
          />
          <div
            class="feature-card__center-container"
          >
            <div
              class="section__outer section__outer--inherit-color feature-card__title"
            />
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`<NavigationItem /> Should render correctly hrefExternal 1`] = `
<div>
  <a
    class="card feature-card card--atomic card--not-disabled"
    href="https://duckduckgo.com/"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      >
        <div
          aria-hidden="true"
          class="feature-card__outer-container feature-card__outer-container--auto"
        >
          <div
            class="feature-card__date"
          />
          <div
            class="feature-card__center-container"
          >
            <div
              class="section__outer section__outer--inherit-color feature-card__title"
            />
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`<NavigationItem /> Should render correctly with custom className 1`] = `
<div>
  <a
    class="card feature-card test-custom-classname card--atomic card--not-disabled"
    href="/test"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      >
        <div
          aria-hidden="true"
          class="feature-card__outer-container feature-card__outer-container--auto"
        >
          <div
            class="feature-card__date"
          />
          <div
            class="feature-card__center-container"
          >
            <div
              class="section__outer section__outer--inherit-color feature-card__title"
            />
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`<NavigationItem /> Should render with nextified href 1`] = `
<div>
  <a
    class="card feature-card card--atomic card--not-disabled"
    href="/blog/test"
  >
    <div
      class="card__outer-container"
    >
      <div
        class="card__background"
      />
      <div
        class="card__content-container card__content-container--padded"
      >
        <div
          aria-hidden="true"
          class="feature-card__outer-container feature-card__outer-container--auto"
        >
          <div
            class="feature-card__date"
          />
          <div
            class="feature-card__center-container"
          >
            <div
              class="section__outer section__outer--inherit-color feature-card__title"
            />
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
